ext/sqlite3/sqlite3.c

```diff
@@ -591,17 +593,19 @@ PHP_METHOD(SQLite3, query)
 	ZVAL_OBJ(&result->stmt_obj_zval, Z_OBJ(stmt));
 
 	return_code = sqlite3_step(result->stmt_obj->stmt);
+	stmt_obj->pending_step_return_code = return_code;
 
 	switch (return_code) {
-		case SQLITE_ROW: /* Valid Row */
 		case SQLITE_DONE: /* Valid but no results */
+			sqlite3_reset(stmt_obj->stmt);
```

Comment:
This fixes the lastErrorCode / Msg test failure - sqlite3_reset docs mentions something about copying the error code from the call into the database object. I suppose the same thing should be done in stmt execute, probably with another test written for errors occuring through prepared statements rather than queries like this.

Something slightly magical here, this doesn't cause duplicate query execution when fetchArray is called on the result set of an INSERT query because we still set the "pending step" flag and return false on the first call to fetchArray.